woocommerce/woocommerce-android
# Removing extra onboarding call in case of onboarding check ends up with an error

Closes: #6416 


### Description

The PR implements a small optimisation. In onboarding error case we pass the onboarding state and show the error state right away, without refetching it again

### Testing instructions

* Try happy payment flow and notice that there are not changes
* Try different error cases (Pending requirements, a country not supported). Notice that error state is shown right away and network call is not performed

### Images/gif


https://user-images.githubusercontent.com/4923871/166422664-a3f6b873-f15a-4e92-a60b-3d945438cf0e.mp4
